challenges/2023-08-08-how-much-water/challenge.md

```diff
@@ -0,0 +1,62 @@
+
+You are given a list of integers, where each value represents the height of a vertical bar that can be placed perpendicular to the x-axis and the index of each value represents it's position on the x-axis. Using this list, find the maximum amount of water that can be contained by any two bars together with the x-axis. For example, given the list `[1, 8, 2, 6, 2, 5, 4, 8, 3, 7]`, the maximum amount of water that can be contained is `49`, like so:
+
+
+```
+     [  1,    8,    2,    6,    5,    4,    8,    3,    7]
+
+     |                                                      
+ 8 - |        #                             #               
+     |        #                             #                 ___
+ 7 - |        #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~#~~~~~~~~~~~#      |
+     |        #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~#~~~~~~~~~~~#      |
+ 6 - |        #~~~~~#~~~~~~~~~~~~~~~~~~~~~~~#~~~~~~~~~~~#      |
+     |        #~~~~~#~~~~~~~~~~~~~~~~~~~~~~~#~~~~~~~~~~~#      |
+ 5 - |        #~~~~~#~~~~~~~~~~~#~~~~~~~~~~~#~~~~~~~~~~~#      |
+     |        #~~~~~#~~~~~~~~~~~#~~~~~~~~~~~#~~~~~~~~~~~#      |
+ 4 - |        #~~~~~#~~~~~~~~~~~#~~~~~#~~~~~#~~~~~~~~~~~#      7
+     |        #~~~~~#~~~~~~~~~~~#~~~~~#~~~~~#~~~~~~~~~~~#      |
+ 3 - |        #~~~~~#~~~~~~~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#      |
+     |        #~~~~~#~~~~~~~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#      |
+ 2 - |        #~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#      |
+     |        #~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#      |
+ 1 - |  #     #~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#      |
+     |  #     #~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#~~~~~#      |
+ 0 - |------------------------------------------------------  ___
+        |     |     |     |     |     |     |     |     |
+        0     1     2     3     4     5     6     7     8
+
+              |                                         |
+              |-------------------7---------------------|
+              |                                         |
+```
+
+## Rules
+
+- We're not tilting the fish tank, so only calculate the water value assuming the tank is perfectly level.
+- Your input will always be a list containing at least two values.
+- Why do the bars need to be placed in the order given? Because that's what they showed in the TikTok, okay?
+
+## Examples
+
+```
+maximum_water([1, 8, 2, 6, 5, 4, 8, 3, 7]) -> 49
+maximum_water([1, 2, 5, 5, 2, 1])          -> 5
+maximum_water([1, 2, 2, 5, 5, 2, 2, 1])    -> 8
```

Comment:
Isn't `10` the answer for this one? 5 spaces between the first and last 2.

Reply:
You're absolutely right. The one above it is wrong too (2 * 3 == 6). I caught it in the test cases, but missed it in the description. I've fixed them both, now.